vital/types/detected_object_set.cxx

```diff
@@ -126,6 +127,18 @@ add( detected_object_sptr object )
 
 
 // ------------------------------------------------------------------
+void
+detected_object_set::
+add( detected_object_set_sptr detections )
+{
+  VITAL_FOREACH( auto detection, detections->select() )
```

Comment:
It would be more efficient to use:

    m_detected_objects.insert( m_detected_objects.end(), detections.begin(), detections.end() );